draft-ietf-mls-architecture.md

```diff
@@ -815,14 +816,22 @@ MLS requires the following policies to be defined, which restrict the set of
 acceptable behavior in a group. These policies must be consistent between
 deployments for them to interoperate:
 
+- A policy on which ciphersuites are acceptable.
+- A policy on any mandatory or forbidden MLS extensions.
 - A policy on when to send proposals and commits in plaintext instead of
   encrypted.
 - A policy for which proposals are valid to have in a commit, including but not
   limited to:
   - When a member is allowed to add or remove other members of the group.
   - When, and under what circumstances, a reinitialization proposal is allowed.
-  - When proposals from external senders are allowed.
-  - When external joiners are allowed.
+  - When proposals from external senders are allowed and how to authorize
+    those proposals.
+  - When external joiners are allowed and how to authorize those external
+    commits.
+  - Which other proposal types are allowed.
+- A policy of when members should commit pending proposals in a group.
```

Comment:
The protocol doc specifies this to some extent (ie, you must commit proposals before you send an application message). To the extent that this behavior is not specified in the protocol doc, I don't believe further policy is necessary for interop

Reply:
This requires a policy. The protocol only says what clients have to do to make the protocol work, not what is desirable operationally. 
If you have a group with 5000 members and one member tries to leave the group by sending a Remove Proposal for itself, which client or clients Commit(s) the proposal? Do they wait indefinitely for one of them to send an application message (and then delay the message by having to send a Commit first)? Do they all immediately sending a Commit effectively causing a DoS attack on the DS? Does one member or a small number of them try the Commit?  When? According to what algorithm?
This is a serious operational concern.

Reply:
I don't agree that clients doing different things will necessarily work fine here. A client that naively sends a Commit instantly whenever it sees a pending proposal in a large group will not be allowed in some MLS networks.